Add no-side-effects version of navigateToAppLink:error: and navigate:

biasedbit opened the pull request:
_Rehash of #26_

### Summary

- Allows app to know the type of navigation that would result of calling `BFAppLinkNavigation`'s `+navigateToAppLink:error:` or `-navigate:` without having any side effects (i.e. no navigation)
- Also adds a test to ensure same output as `+navigateToAppLink:error:` without the side effects.

### Motivation

Relying on `-navigate:` completely hands off app behavior to Bolts. These no-side-effects functions allow apps to know what behavior Bolts is going to adopt with a given `BFAppLink`.

A developer might want to:

- completely hand off behavior if it's an AppLink with eligible targets or;
- use one of their own web views if it's just a regular link or a link that cannot be handled by any installed apps.

nlutsenko commented:
@biasedbit, looks like this is failing tests right now.
It needs a single addition to the Info.plist of a test target, since we are building and testing on iOS 9 SDK:
```
<key>LSApplicationQueriesSchemes</key>
<array>
    <string>bolts</string>
</array>
```

biasedbit commented:
I forked the project from most recent master, shouldn't that be in the project already? I guess my question is if the tests were working before and I haven't touched anything other than code files, why would I have to change my fork's project file to include that?

(I'll try adding that and let you know how it went but I really am curious as to why it's necessary for this PR.)

biasedbit commented:
I can't seem to get the tests to run locally, or the project to even build for that matter. I've ran `git submodule update --init --recursive`, which pulled in some stuff into `Vendor` but I'm still getting warnings that files can't be found and build errors:

<img width="375" alt="screen shot 2016-04-07 at 15 31 19" src="https://cloud.githubusercontent.com/assets/337352/14364234/f6e3c7ee-fcd5-11e5-8f91-7dcaa06e1e38.png">

Is there anything else I need to do to get the project to run/test locally? Couldn't find any info on the README.

nlutsenko commented:
The change is required because we are running with iOS 9 SDK and you are required to specify which schemes you are querying for there - in this case it is `bolts`.

In regards to the test failure - after checking out all the submodules - make sure you clean the Derived Data and relaunch Xcode. It doesn't play nicely if you were missing configs and now have them.

biasedbit commented:
Oh, ofc! I'm using `canOpenURL:`, not `openURL:`, makes sense.

Thanks for the tip on the derived data and the hand-holding — I was completely out of context 👍 

Anyway, tests are successful locally. Rebased, squashed and pushed.
